hybrisprovider.cpp

```diff
@@ -1387,11 +1392,15 @@ bool HybrisProvider::positioningEnabled()
     m_agpsEnabled = (agpsAgreementAccepted || oldAgpsAgreementAccepted) && (agpsEnabled || oldAgpsEnabled);
     m_agpsOnlineEnabled = agpsOnlineEnabled || (oldAgpsAgreementAccepted && oldAgpsEnabled);
 
-    // check the keys related to the location and gps enablement, plus gps power state
+    // check the keys related to the location and gps enablement.
+    // also update DeviceControl if necessary (expose new gps enablement state to connman).
     bool locationEnabled = settings.value(LocationSettingsEnabledKey, false).toBool();
```

Comment:
and todo: change this entire function to use LocationSettings instead of QSettings or libsailfishkeyprovider directly, since we need to perform writes.  I should also update LocationSettings so that it takes the process mutex into use, to avoid write collisions between geoclue-hybris (i.e., triggered by connman) and MDM and jolla-settings.